### PR TITLE
Fix mr_test registration + curl failures on EC2

### PR DESCRIPTION
**Pull request by lilyeyes**
Fix mr_test registration + curl failures on EC2

Fix mr_test registration failed on 'zypper lr' reports 'Warning: No repositories defined'
Fix mr_test qesap_terraform failed on 'curl: option -o: requires parameter'
Also increased the timeout value as https://openqaworker15.qa.suse.cz/tests/114030 (failed on timed out, it is not related to my changes)
Also remove the `space` after `public_key` for all files under `data/sles4sap/qe_sap_deployment`

TEAM-7137 - Fix all "FLAVOR=EC2-SAP-PAYG-Incidents-saptune" test cases failed on 'registration'

Related ticket: https://jira.suse.com/browse/TEAM-7137
Needles: NA
Verification runs:
https://openqaworker15.qa.suse.cz/tests/114028 (passed)
https://openqaworker15.qa.suse.cz/tests/114040 (failed on "Test died: 'qesap.py ansible' return: 2" it seems not related to my changes)

### CODE REVIEW

File: tests/publiccloud/registration.pm [hunk 1]
```diff
@@ -26,7 +26,7 @@ sub run {
 
     select_host_console();    # select console on the host, not the PC instance
 
-    registercloudguest($args->{my_instance}) if (is_byos());
+    registercloudguest($args->{my_instance}) if (is_byos() || get_var('PUBLIC_CLOUD_FORCE_REGISTRATION'));
```
Comment (grisu48) [inline tests/publiccloud/registration.pm:29]: If we're introducing a new variable, please add it also to [variables.md](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/variables.md)

Comment (lilyeyes) [reply tests/publiccloud/registration.pm:29]: > If we're introducing a new variable, please add it also to [variables.md](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/variables.md)

Done. thanks for the reminder :).

**Review by pdostal — APPROVED** [review]

**Comment by pdostal** [comment]
Just adding Anton as he might be interested.

File: data/sles4sap/qe_sap_deployment/mr_test_ec2.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ terraform:
     # GENERAL VARIABLES #
     aws_region: "%PUBLIC_CLOUD_REGION%"
     admin_user: "cloudadmin"
-    public_key : "~/.ssh/id_rsa.pub"
+    public_key: "~/.ssh/id_rsa.pub"
```
Comment (mpagot) [inline data/sles4sap/qe_sap_deployment/mr_test_ec2.yaml:9]: I can see the extra space present in other conf.yaml

```
data/sles4sap/qe_sap_deployment/hanasr_azure.yaml
data/sles4sap/qe_sap_deployment/hanasr_ec2.yaml
data/sles4sap/qe_sap_deployment/hanasr_gcp.yaml
data/sles4sap/qe_sap_deployment/mr_test_azure.yaml
data/sles4sap/qe_sap_deployment/mr_test_ec2.yaml
data/sles4sap/qe_sap_deployment/sles4sap_azure_generic.yaml
```

do you like to remove it from there too?

**Review by mpagot — APPROVED** [review]
LGTM

File: data/sles4sap/qe_sap_deployment/mr_test_ec2.yaml (hunk 2 above)
Comment (lilyeyes) [reply data/sles4sap/qe_sap_deployment/mr_test_ec2.yaml:9]: Done.  But I don't know why there are those changes, I really did nothing on `SAPCONF: true`.
    - SAPCONF: true
    + SAPCONF: true

Comment (asmorodskyi) [reply data/sles4sap/qe_sap_deployment/mr_test_ec2.yaml:9]: before this commit your files didn't had empty line at the end of file

**Review by asmorodskyi — COMMENTED** [review]
LGTM

**Pull request merged**